### ISSUE TITLE
How do we differentiate between DOM and state?

### ISSUE THREAD

**Issue by DaveOrDead** [fedordead/vanilla-ui#25]
At the moment we have the following set up:

```
    // Stores all the constant dom nodes for the component regardless of instance.
    let DOM = {
        dialogs: qa(dialog),
        page: qa('body')
    };

    // Keeps track of current instance of dialog.
    let state = {
        currentOpenButton: null,
        currentDialog: null,
        focusableElements: null
    };
```

This could potentially cause confusion as `currentOpenButton` and `currentDialog` are technically also dom nodes but living in state because they only belong to that instance.

I quite like having the two distinct objects personally, but if this is a convention we're going to have across all components we should probably agree on it.

**Comment by chris-pearce** [comment]
Sorry I'm really stumped on this esp when my JS skills aren't mega. Maybe putting it to the rest of the FED || DEAD group in Slack?

**Comment by bnhovde** [comment]
I also like having them as two objects. Perhaps the `current` part of the names are redundant as it's part of the component state? Perhaps `state.button` and `state.dialog` are good enough?

One thing I've been doing lately that may or may not be common practice is ensure all variables that points to a DOM element is named `someEl` or similar. It's helped me in the past when looking through code. Without that it's hard to tell whether `openButton` is an element or a function.

Saying that, since we store our elements under the `DOM` namespace, so it may not be so critical here.

**Comment by DaveOrDead** [comment]
yeah, all the ones in `DOM.` namespace will be elements so may be redundant.  But the ones in `state.` could be a mix of elements and general info.  e.g the Typeahead has `state.isDropdownOpen: false`

But then it would be weird having different conventions between those 2 objects.

Although now I think about it, I'm not sure what else would be inside the more "global" DOM namespace.  Currently it'd just be body or root / html.  Which I was saying to @chris-pearce yesterday might need to get pulled out so they can be shared across components
